server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -60,6 +60,7 @@ public class RemoteFsTranslog extends Translog {
     private final BooleanSupplier primaryModeSupplier;
     private final RemoteTranslogTransferTracker remoteTranslogTransferTracker;
     private volatile long maxRemoteTranslogGenerationUploaded;
+    private final Object uploadMutex = new Object();
```

Comment:
lets add java doc on why we need this .  Preferably in `performUpload` .

Reply:
Done